### PR TITLE
Fixed Comment Button 

### LINKED ISSUE #402: [BUG] Cannot add comments to the post.
- **Issue Description:** Cannot add comments to the post ,gives exception.
- **Issue Severity:** Medium
- **Environment Details:**   createComment() method in [newsArticle.dart](https://github.com/PalisadoesFoundation/talawa/blob/master/lib/views/pages/newsfeed/newsArticle.dart)
- **Observed Behaviour:** Does not add comment.
- **Expected Behaviour:** Should add comment and display a toast regarding comment status.
- **Steps to reproduce issue:** Go to the Newsfeed page and try adding comment to a post
- **Snapshots/Videos:**  

https://user-images.githubusercontent.com/57677520/111756791-9f248080-88c0-11eb-9892-8243df16c0c1.mp4

- **Does this issue need immediate attention?** Yes, for testing comments. And testing #399 
- **Are you willing to work on this issue:** Yes , I will send a patch immediately.
- **Other information:**  Also there are few unnecessary calls to functions in the same file which can be removed.

### PR DESCRIPTION
**Pull request by utkarshshendge**
**What kind of change does this PR introduce?**

This PR fixes #402  that is it restores comment functionality , shows toast to display comment status to user .

**Did you add tests for your changes?**
No

**If relevant, did you update the documentation?**
No

**Summary**
Before:

https://user-images.githubusercontent.com/57677520/111770895-00ece680-88d1-11eb-8fe4-62ce4a124305.mp4

After:

https://user-images.githubusercontent.com/57677520/111772574-04816d00-88d3-11eb-897b-027770bcda89.mp4

**Does this PR introduce a breaking change?**
No

### CODE REVIEW

File: lib/views/pages/newsfeed/newsArticle.dart [hunk 1]
```diff
@@ -103,32 +110,32 @@ class _NewsArticleState extends State<NewsArticle> {
...
-                          createComment();
-                        },
-                      ),
-                      hintText: 'Leave a Comment....',
-                      border: OutlineInputBorder(
-                          borderRadius: BorderRadius.circular(20.0),
-                          borderSide: BorderSide(color: Colors.teal))),
-                  controller: commentController,
-                ),
+                  child: TextField(
```
Comment (Sagar2366) [inline lib/views/pages/newsfeed/newsArticle.dart:118]: @utkarshshendge is this change required to fix the issue ?

Comment (utkarshshendge) [reply lib/views/pages/newsfeed/newsArticle.dart:118]: No @Sagar2366 Sir, looks like its result of whitespace I gave.
Should I create another commit?

**Comment by yasharth291** [comment]
@utkarshshendge Why are you removing multiline comment limit?

**Comment by utkarshshendge** [comment]
> @utkarshshendge Why are you removing multiline comment limit?

Sorry @yasharth291 , you are referring to  which  multiline comment limit?

**Comment by yasharth291** [comment]
minline and maxline you have commented that.

**Comment by utkarshshendge** [comment]
@yasharth291 , I didn't do that .
Check:
https://github.com/PalisadoesFoundation/talawa/blob/master/lib/views/pages/newsfeed/newsArticle.dart
its already there.

**Comment by yasharth291** [comment]
oh sorry so can you uncomment them plz

File: lib/views/pages/newsfeed/newsArticle.dart (hunk 1 above)
Comment (Sagar2366) [reply lib/views/pages/newsfeed/newsArticle.dart]: Yes, it clearly shows the effect of this in second video.

**Comment by Sagar2366** [comment]
@utkarshshendge try for below testcase:
`Hi
Thank you for contributing to Talawa project`
It does not add comment if comment is multiline.

Second problem is keyboard stay open even after adding comment. I t should get closed immediately once comment is added.

**Comment by utkarshshendge** [comment]
@Sagar2366   Sir, this PR fixes 2nd problem as well.
Checking for the first one..

**Comment by utkarshshendge** [comment]
Hello @Sagar2366  Sir,
Added Multiline comment functionality.

https://user-images.githubusercontent.com/57677520/111864966-14a85380-898a-11eb-8fe5-c8d5da2f3d81.mp4

**Comment by Sagar2366** [comment]
@utkarshshendge good work.
Can you add functionality to update comment count outside on newsfeed page also as part of this PR ?

**Comment by utkarshshendge** [comment]
Yes @Sagar2366  Sir, currently we have to pull down to refresh the comment count, expected behaviour is that it should automatically update the comment count. I will add it to this PR.

**Comment by yasharth291** [comment]
@utkarshshendge can you please tell what is comment flag?

**Comment by utkarshshendge** [comment]
@Sagar2366 Sir, Now the comments are updated on newsfeed page.

https://user-images.githubusercontent.com/57677520/111905684-ea7e9080-8a72-11eb-81de-648d01a4c0dc.mp4

BUT: Found one more issue ,and looks like we need to create seperate issue for it.
Issue:
 We are not able to call the onWillPop() method .
I found this for all children pages of BottomNavigationBar.
And without it we wont be able to override the back button function in  app.
As an effect of this comment number wont be updated if you press back button for navigating to newsfeed page.
I spent few hours trying to find a work-around for this ,but nothing seemed to work.

**Comment by yasharth291** [comment]
can you please tell what is comment flag

**Comment by utkarshshendge** [comment]
> @utkarshshendge can you please tell what is comment flag?

@yasharth291 
Flags are simply boolean values created to restrict certain operations.
As a result of isCommentAdded (Flag) newsfeed wont be updated if comments are not added.
Hope this helps.

**Comment by yasharth291** [comment]
okay thanks @utkarshshendge

**Comment by Sagar2366** [comment]
@utkarshshendge not sure if you have noticed already and planning to fix it as part of this PR:
1. After adding new comment to news post, we click on load comments option to see all comments including new comment.  For a while it shows 0 comments and after few seconds it updates comment count
2. On returning back to main newsfeed page after adding new comment, it does not update comment count and we need to pull down screen to refresh the comment count.

Can you please fix both of these issues ? 
If you think it will be too much of work, create another issue and work on it. [Note: Share your thoughts on this before creating new issue]
Please add unit and widget tests for your work in this PR.

**Comment by utkarshshendge** [comment]
@Sagar2366  Sir,
         
1. It takes time because we are fetching the comments from server, we can add a toast to display status.
         
 2. I already fixed it but there's a catch to it please see. https://github.com/PalisadoesFoundation/talawa/pull/405#issuecomment-803575689 

I will add tests. But I think it will take some time.

**Review by Sagar2366 — APPROVED** [review]

**Comment by Sagar2366** [comment]
@utkarshshendge can you pull latest changes from master to fix travis CI failing issue.

**Comment by utkarshshendge** [comment]
> @utkarshshendge can you pull latest changes from master to fix travis CI failing issue.

Done @Sagar2366  Sir.

**Pull request merged**